frame/dex/src/lib.rs

```diff
@@ -413,9 +424,22 @@ pub mod pallet {
 
 				T::PoolAssets::burn_from(pool.lp_token, &pallet_account, liquidity)?;
 
-				T::Assets::transfer(asset1, &pallet_account, &withdraw_to, amount1, false)?;
-				T::Assets::transfer(asset2, &pallet_account, &withdraw_to, amount2, false)?;
-
+				match asset1 {
+					MultiAssetId::Native => {
+						T::Currency::transfer(&pallet_account, &withdraw_to, amount1, false)?;
+					},
+					MultiAssetId::Asset(asset1) => {
+						T::Assets::transfer(asset1, &pallet_account, &withdraw_to, amount1, false)?;
+					},
+				}
+				match asset2 {
+					MultiAssetId::Native => {
+						T::Currency::transfer(&pallet_account, &withdraw_to, amount2, false)?;
+					},
+					MultiAssetId::Asset(asset2) => {
+						T::Assets::transfer(asset2, &pallet_account, &withdraw_to, amount2, false)?;
+					},
+				}
```

Comment:
Could pull these out into a transfer function that takes a MultiAsset.

Reply:
yeah, I think it's a good idea